File: project/jimmer-core/src/test/java/org/babyfish/jimmer/model/SamePassword.java
package org.babyfish.jimmer.model;

import javax.validation.Constraint;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
@Constraint(validatedBy = SamePasswordValidator.class)
public @interface SamePassword {

    String message() default "Two passwords must be same";
}

File: project/jimmer-core-kotlin/src/test/kotlin/org/babyfish/jimmer/kt/model/SamePassword.kt
package org.babyfish.jimmer.kt.model

import javax.validation.Constraint

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
@Constraint(validatedBy = [RegisterRequestValidator::class])
annotation class SamePassword(
    val message: String = "The passwords must be same"
)